### PR TITLE
Modify rule S2272: reworked rule description for LaYC format

### PR DESCRIPTION
**Pull request by kaufco**

### CODE REVIEW

File: rules/S2272/java/rule.adoc [hunk 1]
```diff
@@ -1,10 +1,13 @@
 == Why is this an issue?
 
-By contract, any implementation of the `java.util.Iterator.next()` method should throw a `NoSuchElementException` exception when the iteration has no more elements.
-Any other behavior when the iteration is done could lead to unexpected behavior for users of this `Iterator`.
+The `java.util.Iterator.next()` method must throw a `NoSuchElementException` when there are no more elements in the iteration.
+Any other behavior is non-compliant with the API contract and may cause unexpected behavior for users.
 
+== How to fix it
```
Comment (irina-batinic-sonarsource) [inline rules/S2272/java/rule.adoc:7]: Usually, if we add a `How to fix it` section, we add some text here.
IMO, for this particular case, we can skip it.

File: rules/S2272/java/rule.adoc [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 
-=== Noncompliant code example
+=== Code examples
+
+==== Noncompliant code example
 
 [source,java,diff-id=1,diff-type=noncompliant]
 ----
```
Comment (irina-batinic-sonarsource) [inline rules/S2272/java/rule.adoc:13]: If I remember correctly, the code in noncompliant and compliant sections should compile.
Let's remove `...`

File: rules/S2272/java/rule.adoc [hunk 3]
```diff
@@ -19,8 +22,7 @@ public class MyIterator implements Iterator<String> {
 }
 ----
 
-
-=== Compliant solution
+==== Compliant solution
 
 [source,java,diff-id=1,diff-type=compliant]
```
Comment (irina-batinic-sonarsource) [inline rules/S2272/java/rule.adoc:27]: Same here. Let's remove `...`

File: rules/S2272/java/rule.adoc [hunk 4]
```diff
@@ -37,7 +39,9 @@ public class MyIterator implements Iterator<String> {
 
 == Resources
 
-* https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#next--[JavaDoc 8 - java.util.Iterator]
+=== Documentation
+
+* https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#next--[Java SE 8 API Specification: Iterator]
```
Comment (irina-batinic-sonarsource) [inline rules/S2272/java/rule.adoc:44]: Let's reference the latest Java version: https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/Iterator.html#next()

**Review by irina-batinic-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S2272/java/rule.adoc (hunk 2 above)
Comment (kaufco) [reply rules/S2272/java/rule.adoc:9]: Fixed.

File: rules/S2272/java/rule.adoc (hunk 1 above)
Comment (kaufco) [reply rules/S2272/java/rule.adoc]: Fixed.

File: rules/S2272/java/rule.adoc (hunk 3 above)
Comment (kaufco) [reply rules/S2272/java/rule.adoc:22]: Fixed.

File: rules/S2272/java/rule.adoc (hunk 4 above)
Comment (kaufco) [reply rules/S2272/java/rule.adoc:38]: Resolved (not fixed).

Comment (irina-batinic-sonarsource) [reply rules/S2272/java/rule.adoc:38]: I can't entirely agree it is resolved.
As was explained in [this slack thread](https://sonarsource.slack.com/archives/C04P0SB2KJP/p1692031949269499?thread_ts=1692028946.548869&cid=C04P0SB2KJP) let's go with the lowest supported version.

**Review by irina-batinic-sonarsource — APPROVED** [review]

**Pull request merged**